tests/integration/otlp/test_base_sw_otlp.py

```diff
@@ -0,0 +1,219 @@
+
+    def _assert_reporter(self, reporter):
+        assert isinstance(reporter, NoOpReporter)
+
+    def _setup_test_traces_export(
+        self,
+        apm_config,
+        configurator,
+        reporter,
+    ):
+        oboe_api = OboeAPI()
+        super()._setup_test_traces_export(
+            apm_config,
+            configurator,
+            reporter,
+            oboe_api,
+        )
+
+    def _setup_test_metrics_reader(self):
+        # Set InMemory reader instead of exporter
+        self.metric_reader = InMemoryMetricReader()
+        meter_provider = MeterProvider(
+            metric_readers=[self.metric_reader],
+        )
+        metrics_api.set_meter_provider(meter_provider)
+
+    def _setup_test_logs_export(self):
+        self.logger_provider = LoggerProvider()
+
+        # Set InMemory exporter for testing generated telemetry
+        # instead of SolarWinds/OTLP exporter
+        self.memory_log_exporter = InMemoryLogExporter()
+        self.logger_provider.add_log_record_processor(
+            SimpleLogRecordProcessor(self.memory_log_exporter)
+        )
+        logger = logging.getLogger("foo-logger")
+        logger.propagate = False
+        logger.addHandler(
+            LoggingHandler(logger_provider=self.logger_provider)
+        )
+
+    def setUp(self):
+        super().setUp()
+        self._setup_test_metrics_reader()
+        self._setup_test_logs_export()
+
+        # Finish setup by running Flask app instrumented with APM Python
+        self._setup_instrumented_app()
+
+    def assertFlaskMetrics(self):
+        """Assert server metrics generated from test app Flask instrumentation"""
+        metric_data = self.metric_reader.get_metrics_data()
+        metrics = metric_data.resource_metrics[0].scope_metrics[0].metrics
+        self.assertEqual(len(metrics), 2)
+        active_requests = metrics[0]
+        self.assertEqual(
+            active_requests.name,
+            "http.server.active_requests",
+        )
+        self.assertEqual(
+            active_requests.data.data_points[0].attributes,
+            {"http.request.method": "GET", "url.scheme": "http"},
+        )
+        request_duration = metrics[1]
+        self.assertEqual(
+            request_duration.name,
+            "http.server.request.duration",
+        )
+        self.assertEqual(
+            request_duration.data.data_points[0].attributes,
+            {
+                "http.request.method": "GET",
+                "url.scheme": "http",
+                "network.protocol.version": "1.1",
+                "http.response.status_code": 200,
+                "http.route": "/test_trace/"
+            },
+        )
+
+    def assertRequestsMetrics(self, status_code=True):
+        """Assert client metrics from test app Requests instrumentation.
+        Instrumentor sometimes does not include status_code attribute,
+        i.e. decision is not-sampled."""
```

Comment:
minor aside re: status_code only available if span is sampled, seems a bug in upstream.

Reply:
!!!! It might be this!! https://github.com/open-telemetry/opentelemetry.io/issues/6274